runtimes/protocol/notification.ts

```diff
@@ -0,0 +1,69 @@
+import { MessageType, ProtocolNotificationType } from './lsp'
+
+export interface NotificationContent {
+    readonly text: string
+    readonly title?: string
+}
+
+export namespace FollowupNotificationActionType {
+    export const Acknowledge = 'Acknowledge'
+}
+
+export type FollowupNotificationActionType = typeof FollowupNotificationActionType.Acknowledge
+
+export namespace NotificationActionType {
+    export const Url = 'Url'
+    export const Marketplace = 'Marketplace'
+}
+
+export type NotificationActionType =
+    | typeof NotificationActionType.Url
+    | typeof NotificationActionType.Marketplace
+    | typeof FollowupNotificationActionType.Acknowledge
+
+export interface NotificationAction {
+    readonly text: string
+    readonly type: NotificationActionType
+}
+
+export interface UrlAction extends NotificationAction {
+    readonly type: typeof NotificationActionType.Url
+    readonly url: string
```

Comment:
nit:  LSP spec uses "uri" instead of "url" throughout.  Consider changing for consistency.

Reply:
We already use both options throughout the repo.